helix-core/src/syntax/span.rs

```diff
@@ -67,6 +71,79 @@ pub fn span_iter(spans: Vec<Span>) -> impl Iterator<Item = HighlightEvent> {
     }
 }
 
+impl SpanIter {
+    fn start_span(&mut self, span: Span) {
+        use HighlightEvent::*;
+
+        debug_assert!(span.start <= span.end);
+        self.event_queue
+            .push_back(HighlightStart(Highlight(span.scope)));
+        self.range_ends.push(span.end);
+    }
+
+    fn process_range_end(&mut self, end: usize) -> HighlightEvent {
```

Comment:
another minor nit: I think this would be clearer as `emit_range_end` since it is either queueing or directly returning the ending event

Reply:
fixed in 20302313b5d274d231c8c2e505a5306601c0173d